src/audio_unit/looper.rs

```diff
@@ -88,24 +110,71 @@ impl Looper {
                 }
             }
             Playing { position, total } => {
+                self.process_samples_playing(position, total, input, output);
+            }
+            PlayingAwaitingOverdub { position, total } => {
+                self.process_samples_playing(position, total, input, output);
+            }
+            Overdubbing { position, total } => {
                 let next_position = position + output.len();
 
                 if next_position <= total {
                     output.copy_from_slice(&self.buffer[position..next_position]);
+
+                    for (sample_number, input_sample) in input.iter().enumerate() {
```

Comment:
Are `input` and `output` guaranteed to have the same length (doesn't affect the logic here but I was wondering this while I was thinking through it)?

Reply:
I think that's guaranteed by the audio library we're using, but I'm not positive. (It's certainly been true in all the tests I've done.)